# HG changeset patch
# User Wohlstand <admin@wohlnet.ru>
# Date 1516944267 -10800
#      Fri Jan 26 08:24:27 2018 +0300
# Node ID 8e028866d092e4e741a2d84ebfb080f303eeade7
# Parent  b0e4f5085af4398f8e0ceec06e6d0fe035734007
Added ability to force disable WASAPI from CMake build

diff -r b0e4f5085af4 -r 8e028866d092 CMakeLists.txt
--- a/CMakeLists.txt	Tue Feb 13 22:58:20 2018 -0800
+++ b/CMakeLists.txt	Fri Jan 26 08:24:27 2018 +0300
@@ -1208,7 +1208,8 @@
       set(SOURCE_FILES ${SOURCE_FILES} ${DSOUND_AUDIO_SOURCES})
     endif()
 
-    if(HAVE_AUDIOCLIENT_H AND HAVE_MMDEVICEAPI_H)
+    option(ENABLE_WASAPI_SUPPORT "Use WASAPI when it is available" ON)
+    if(ENABLE_WASAPI_SUPPORT AND HAVE_AUDIOCLIENT_H AND HAVE_MMDEVICEAPI_H)
       set(SDL_AUDIO_DRIVER_WASAPI 1)
       file(GLOB WASAPI_AUDIO_SOURCES ${SDL2_SOURCE_DIR}/src/audio/wasapi/*.c)
       set(SOURCE_FILES ${SOURCE_FILES} ${WASAPI_AUDIO_SOURCES})
